src/python/pants/backend/codegen/protobuf/java/java_protobuf_library.py

```diff
@@ -32,15 +36,9 @@ def __init__(self, payload=None, buildflags=None, imports=None, **kwargs):
       'import_specs': PrimitiveField(imports or ())
     })
     super(JavaProtobufLibrary, self).__init__(payload=payload, **kwargs)
-    if buildflags is not None:
-      logger.warn("Target definition at {address} sets attribute 'buildflags' which is "
-                  "ignored and will be removed in a future release"
-                  .format(address=self.address.spec))
-
-  @classmethod
-  def imported_jar_library_spec_fields(cls):
-    """Fields to extract JarLibrary specs from.
 
-    Required to implement the ImportJarsMixin.
-    """
-    yield ('imports', 'import_specs')
+    deprecated_conditional(
+      lambda: buildflags is not None,
+      '1.16.0.dev1',
+      "Target definition at {addr} setting attribute 'buildflags'".format(addr=self.address.spec),
```

Comment:
This should say what the alternative is.

Reply:
(did the above in 25e1546!)